subkey/src/main.rs

```diff
@@ -255,11 +255,19 @@ fn execute<C: Crypto>(matches: clap::ArgMatches) where
 				panic!("signature is an invalid length. {} bytes is not the expected value of {} bytes", sig_data.len(), sig.as_ref().len());
 			}
 			sig.as_mut().copy_from_slice(&sig_data);
-			let uri = matches.value_of("uri")
+			let hex_uri = matches.value_of("uri")
 				.expect("public uri parameter is required; thus it can't be None; qed");
-			let pubkey = <<C as Crypto>::Pair as Pair>::Public::from_string(uri).ok().or_else(||
+			let uri = if hex_uri.starts_with("0x") {
```

Comment:
bad naming - a `hex_uri` doesn't become a `uri` merely be removing an optional leading `0x`.